src/ia-book-actions.js

```diff
@@ -677,55 +687,77 @@ export default class IABookActions extends LitElement {
   startTimerCountdown() {
     window?.IALendingIntervals?.clearTimerCountdown();
 
-    let secondsLeft = Number(this.lendingStatus.secondsLeftOnLoan);
+    const secondsLeft = Number(this.lendingStatus.secondsLeftOnLoan);
     this.timeWhenTimerStart = new Date();
 
     window.IALendingIntervals.timerCountdown = setInterval(async () => {
-      secondsLeft -= this.timerExecutionSeconds;
-      secondsLeft = Math.round(secondsLeft); // round number
-
-      // re-sync timer if gone off because of background window
-      // side effect: updates this.lendingStatus & kicks off lifecycle,
-      // if really updated, escape from here
-      const resync = await this.reSyncTimerIfGoneOff(secondsLeft);
-
-      if (resync) {
-        log('startTimerCountdown --- stale, timer has resyncd', {
-          secondsLeft,
-        });
-        return;
-      }
+      // interval execution block
+      await this.setIntervalExecutionBlock(secondsLeft);
+    }, this.timerExecutionSeconds * 1000);
+  }
 
-      log('startTimerCountdown --- countdown still valid. continue...', {
-        resync,
+  /**
+   * setInterval execution function do following things
+   * - if setInterval timer gone off, let's resync
+   * - loan-renew attempt to renew the loan OR show relavant message
+   * - if loan has expired, clear interval timers
+   *
+   * @param {Number} secondsLeftOnLoan
+   */
+  async setIntervalExecutionBlock(secondsLeftOnLoan) {
```

Comment:
let's describe this function's behaviour in the name, like, "loanStatusCheckInterval" 😄